lib/imparse.c

```diff
@@ -214,3 +215,39 @@ EXPORTED int imparse_isnumber(const char *s)
     }
     return 1;
 }
+
+/*
+ * Parse a range from the string starting at the pointer pointed to by 's'.
+ * and populate the structure in the pointer at 'range'.
+ * Returns 0 on success, and non-zero on failure.
+ */
+EXPORTED int imparse_range(char *s, range_t *range)
```

Comment:
Would be nice to see a test for this added to cunit/imparse.testc

Reply:
done